app.py

```diff
@@ -557,53 +581,18 @@ def snap_details(snap_name):
             "end": today.strftime('%Y-%m-%d')
         }
     ]
-    metrics_response = _get_from_cache(
-        SNAP_METRICS_URL.format(snap_name=snap_name),
-        headers=METRICS_QUERY_HEADERS,
-        json=metrics_query_json
-    )
-
-    geodata = metrics_response.json()[0]['series']
-
-    # Normalise geodata from API
-    users_by_country = {}
-    max_users = 0.0
-    for country_percentages in geodata:
-        country_code = country_percentages['name']
-        users_by_country[country_code] = {}
-        percentages = []
-        for daily_percent in country_percentages['values']:
-            if daily_percent is not None:
-                percentages.append(daily_percent)
-
-        if len(percentages) > 0:
-            users_by_country[country_code]['percentage_of_users'] = (
-                sum(percentages) / len(percentages)
-            )
-        else:
-            users_by_country[country_code]['percentage_of_users'] = 0
-
-        if max_users < users_by_country[country_code]['percentage_of_users']:
-            max_users = users_by_country[country_code]['percentage_of_users']
 
-    calculate_colors(users_by_country, max_users)
-
-    # Build up country info for every country
-    country_data = {}
-    for country in pycountry.countries:
-        country_info = users_by_country.get(country.alpha_2)
-        percentage_of_users = 0
-        color_rgb = [229, 245, 223]
-        if country_info is not None:
-            percentage_of_users = country_info['percentage_of_users'] or 0
-            color_rgb = country_info['color_rgb'] or [229, 245, 223]
+    metrics_response = get_public_metrics(
+        snap_name,
+        metrics_query_json
+    )
 
-        country_data[country.numeric] = {
-            'name': country.name,
-            'code': country.alpha_2,
-            'percentage_of_users': percentage_of_users,
-            'color_rgb': color_rgb
-        }
+    users_by_country = normalize_metrics(
+        metrics_response[0]['series']
+    )
+    country_data = build_country_info(
+        users_by_country,
```

Comment:
Not sure about our python code style, but these calls seem unnecessarily split into multiple lines as they would easily fit into one line as they don't take many arguments.